cranelift/codegen/src/isa/riscv64/inst/emit.rs

```diff
@@ -1909,48 +1909,6 @@ impl MachInstEmit for Inst {
                 // all zero will cause invalid opcode.
                 sink.put4(0);
             }
-            &Inst::SelectIf {
-                if_spectre_guard: _if_spectre_guard, // _if_spectre_guard not use because it is used to not be removed by optimization pass and some other staff.
-                ref rd,
-                test,
-                ref x,
-                ref y,
-            } => {
-                let label_select_x = sink.get_label();
-                let label_select_y = sink.get_label();
-                let label_jump_over = sink.get_label();
-                let test = allocs.next(test);
-                let x = alloc_value_regs(x, &mut allocs);
-                let y = alloc_value_regs(y, &mut allocs);
-                let rd: Vec<_> = rd.iter().map(|r| allocs.next_writable(*r)).collect();
-                Inst::CondBr {
```

Comment:
`select_spectre_guard` shouldn't emit any branches. While the old code here is wrong, fixing it will require the split between `select` and `select_spectre_guard` again.

Reply:
I don't think we have a way of doing that with the base ISA on RISC-V. Do you know how other compilers handle this?

Reply:
You can use bit masking: https://github.com/RustCrypto/utils/blob/0256fff4d67edb4956f5200b877f41038c4ef7e7/cmov/src/portable.rs#L46-L47

Reply:
Right, but I think that can still be speculated on right? My understanding is that we used `cmov` on x64 and `csel` on AArch64 because we had architectural guarantees that they wouldn't be speculated on?

I'm up for rewriting this if it does prevent speculation though.

Reply:
While it is technically possible, I don't think any cpu would see a benefit in pattern matching this pattern and turning it back into a conditional branch that the branch predictor can speculate.

Reply:
What do you two think is the current state of this PR? I'm not sure what conclusions you've come to from this conversation.

Reply:
I guess this PR is fine for now, but I did like to see a proper implementation of select_spectre_guard in the future.

Reply:
I have a version of this pretty much ready to go, but got distracted by the fuzzer SIMD stuff! I'll clean it up a little bit and push those commits.